/*
 * Copyright 2018- The Pixie Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

syntax = "proto3";

package px.stirling.protocols.http2.testing;

option go_package = "greetpb";

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello(HelloRequest) returns (HelloReply);
  rpc SayHelloAgain(HelloRequest) returns (HelloReply);
}

service StreamingGreeter {
  rpc SayHelloClientStreaming(stream HelloRequest) returns (HelloReply);
  rpc SayHelloServerStreaming(HelloRequest) returns (stream HelloReply);
  rpc SayHelloBidirStreaming(stream HelloRequest) returns (stream HelloReply);
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
  // The number of greetings to return. Only used in streaming method.
  int32 count = 2;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
